crates/rune/src/runtime/runtime_context.rs

```diff
@@ -34,16 +35,20 @@ pub struct RuntimeContext {
     functions: hash::Map<Arc<FunctionHandler>>,
     /// Named constant values
     constants: hash::Map<ConstValue>,
+    /// Deprecation messages for functions
+    deprecations: hash::Map<Box<str>>,
```

Comment:
The deprecation messages should not be associated with the runtime context. We can avoid storing them here and instead have the emitted diagnostics be decorated if a `Unit` (stored in [`DebugInfo`](https://docs.rs/rune/latest/rune/runtime/unit/struct.Unit.html#method.debug_info)) and `Context` is provided.

Else, if they're not available, you'd print a more generic message like:

```
Called deprecated function with hash 0xdeadbeef
```

Note that we have to do something like that regardless (merge deprecation messages from different sources), to support deprecation messages from units which cannot be stored in the runtime context.